### ISSUE TITLE
On Windows, cabal hangs indefinitely when fetching a `git` source-repository-package

### ISSUE THREAD

**Issue by JakobBruenker** [haskell/cabal#8688]
**Describe the bug**
On Windows, when cabal tries fetching a git repo, it hangs indefinitely, in waitForProcess:
https://github.com/haskell/cabal/blob/96dea0ffbace9488f355c6d56a2ab45ff33c3867/Cabal/src/Distribution/Simple/Utils.hs#L783

When run for the first time, it hangs while running `"git.exe" "submodule" "deinit" "--force" "--all"`; if cabal is killed, on subsequent runs it hangs on `"git.exe" "fetch"`.

Running the git command manually in the corresponding working directory does not hang (and after doing so, cabal builds the package successfully).

**To Reproduce**

```
$ git clone https://github.com/JakobBruenker/cabal-hangs
$ cd cabal-hangs/
$ cabal v2-build -v3
```

The last line logged before cabal hangs should be `"C:\Program Files\Git\cmd\git.exe" "submodule" "deinit" "--force" "--all"`, or the equivalent with a different path to `git.exe`.

**Expected behavior**
It should build the package successfully

**System information**
 - Windows 11
 - ghc 9.2.5
 - cabal 3.8.1.0, also tested on HEAD (96dea0f)

**Comment by Mikolaj** [comment]
Thank you for the report. Could this be a `process` package bug? Is this reproduced with the newest `process` on Hackage?

Also, are we sure the same works fine on, say, Linux? Is there a test in CI for exactly this case (e.g., skipped on Windows, because we though it's a CI fluke)?

**Comment by JakobBruenker** [comment]
@Mikolaj I was able to build the same package under WSL, so I believe it does work in Linux, though of course there could be some confounding variable.

Looks when I built from source, cabal used process 1.6.15.0, which is one version older than the newest version on hackage. I can check later today if a small program that just waits for a git process has the same issue on my machine.

**Comment by JakobBruenker** [comment]
@Mikolaj I can confirm that the same thing happens when just using `waitForProcess` with the same commands outside of cabal. I will create an issue in the `process` repo.

**Comment by Mikolaj** [comment]
Well done. Thank you.

**Comment by jasagredo** [comment]
haskell/process#273 is closed by haskell/process#277. Can we close this one? @ulysses4ever pinging you as you reopened this ticket.

**Comment by ulysses4ever** [comment]
Sorry, this was my bad, probably
